src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -68,6 +68,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             If _disposeScanner Then
                 Me._scanner.Dispose()
             End If
+
+            ' Ensure the context's statements have been freed
+            Context?.FreeStatements()
+
+            _pool.Free()
```

Comment:
This is why I had the earlier PR to ensure that the VB parser's dispose is always created